src/silo/storage/metadata_store.cpp

```diff
@@ -43,28 +41,39 @@ unsigned MetadataStore::fill(
    return sequence_count;
 }
 
-void MetadataStore::initializeColumns(
-   const config::DatabaseConfig& database_config,
-   const std::set<std::string>& columns_to_index
-) {
+void MetadataStore::initializeColumns(const config::DatabaseConfig& database_config) {
    for (const auto& item : database_config.schema.metadata) {
       if (item.type == config::DatabaseMetadataType::STRING) {
-         if (columns_to_index.contains(item.name)) {
-            this->indexed_string_columns[item.name] = storage::column::IndexedStringColumn();
-         } else {
-            this->raw_string_columns[item.name] = storage::column::RawStringColumn();
-         }
+         raw_string_columns[item.name] = storage::column::RawStringColumn();
+      } else if (item.type == config::DatabaseMetadataType::INDEXED_STRING) {
+         indexed_string_columns[item.name] = storage::column::IndexedStringColumn();
       } else if (item.type == config::DatabaseMetadataType::PANGOLINEAGE) {
          pango_lineage_columns.emplace(item.name, storage::column::PangoLineageColumn());
       } else if (item.type == config::DatabaseMetadataType::DATE) {
          date_columns.emplace(
             item.name,
-            // TODO(#114) make this configurable
-            item.name == "date" ? storage::column::DateColumn(true)
-                                : storage::column::DateColumn(false)
+            item.name == database_config.schema.date_to_sort_by ? storage::column::DateColumn(true)
+                                                                : storage::column::DateColumn(false)
          );
       }
    }
 }
 
+const storage::column::Column& MetadataStore::getColumn(const config::DatabaseMetadata& metadata
+) const {
+   if (metadata.type == config::DatabaseMetadataType::STRING) {
+      return raw_string_columns.at(metadata.name);
+   }
+   if (metadata.type == config::DatabaseMetadataType::INDEXED_STRING) {
+      return indexed_string_columns.at(metadata.name);
+   }
+   if (metadata.type == config::DatabaseMetadataType::PANGOLINEAGE) {
+      return pango_lineage_columns.at(metadata.name);
+   }
+   if (metadata.type == config::DatabaseMetadataType::DATE) {
+      return date_columns.at(metadata.name);
+   }
+   throw "fill failed";
```

Comment:
Let's not throw a string ;)

Reply:
Ah yes, I added a TODO comment in a following commit, because I was unsure what exception to throw here, as it is more of a helper function. Thinking about it, the way to go is probably returning an `std::optional` instead of callers catching an intermediate exception and then needing to rethrow a more specific exception.

Reply:
An optional sounds like a good idea :+1: 
Optionals and such are usually a good idea in expected error cases that should be recoverable. Exceptions are better suited for unexpected, unrecoverable cases.

Reply:
Well I guess the exception is not recoverable in the path anyways, as the query will be aborted anyways. So I went with a custom exception instead